cmd/gcs/main.go

```diff
@@ -233,6 +236,9 @@ func main() {
 		logrus.WithError(err).Fatal("failed add gcs pid to gcs cgroup")
 	}
 
+	// an OOM event in the GCS control group will likely kill a process hosting
+	// go-routines (or the entire gcs process tree), so there is a chance the
+	// go-routine reporting OOM events to the host will be killed as well
```

Comment:
what are the scenarios in which the entire gcs process tree would be killed?

Reply:
The cgroup [docs](https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v1/memory.html#reclaim) say the OOM killer will kill the bulkiest task in the CGgroup, which I read as all the OS threads backing the Linux GCS. I haven't tested if, when a goroutine exceeded memory limits, there is enough granularity to just kill the one thread it is on, and if that recoverable for the go runtime.

Reply:
I think we have the gcs in it's own cgroup though

Reply:
yup
looking at it again, this comment is poorly worded: if an OOM event happens in gcs's cgroup, `/gcs`, gcs wont be able to catch it, since it will likely be killed off in its entirety, hence why there is no memory limit and its not added to the OOM watcher later on.